test/suite/test_compact07.py

```diff
@@ -0,0 +1,222 @@
+        
+        # Check that the background server is no longer running.
+        stat_cursor = self.session.open_cursor('statistics:', None, None)
+        compact_running = stat_cursor[stat.conn.background_compact_running][2]
+        stat_cursor.close()
+        while compact_running:
+            stat_cursor = self.session.open_cursor('statistics:', None, None)
+            compact_running = stat_cursor[stat.conn.background_compact_running][2]
+            stat_cursor.close()
+    
+        self.assertEqual(compact_running, 0)
+        
+    # FIXME-WT-11450: It's possible for compaction to block the drop table call indefinitely.
+    # We should reenable this test once that situation can be handled by the background compaction
+    # server.
+    
+    # Test the background server while creating and dropping tables.
+    # def test_background_compact_table_ops(self):
+    #     # Trigger the background compaction server first, before creating tables.
+    #     self.session.compact(None,'background=true,free_space_target=1MB')
+
+    #     # Create n tables for background compaction to loop through.
+    #     for i in range(self.n_tables):
+    #         uri = self.uri_prefix + f'_{i}'
+    #         ds = SimpleDataSet(self, uri, self.table_numkv, 
+    #                         key_format=self.key_format, 
+    #                         value_format=self.value_format)
+    #         ds.populate()
+                
+    #     # Delete some data from half of the tables.
+    #     for i in range(self.n_tables):
+    #         if (i % 2 == 0):
+    #             self.delete_range(f'{self.uri_prefix}_{i}')
+                
+    #     # Periodically check that background compaction has done some work on any of the tables.
+    #     while (self.get_files_compacted() == 0):
+    #         time.sleep(1)
+            
+    #     # Now drop all the tables.
+    #     for i in range(self.n_tables):
+    #         self.dropUntilSuccess(self.session, f'{self.uri_prefix}_{i}')
+        
+    # Run background compaction alongside many checkpoints.
+    def test_background_compact_chkpt_stress(self):
+        # FIXME-WT-11399
+        if self.runningHook('tiered'):
+            self.skipTest("this test does not yet work with tiered storage")
+            
+        # Create n tables for background compaction to loop through.
+        for i in range(self.n_tables):
+            uri = self.uri_prefix + f'_{i}'
+            ds = SimpleDataSet(self, uri, self.table_numkv, 
+                            key_format=self.key_format, 
+                            value_format=self.value_format)
+            ds.populate()
+            self.delete_range(uri)
+            
+        self.reopen_conn()
+
+        self.session.compact(None, 'background=true,free_space_target=1MB')    
+        for i in range(1000):
+            self.session.begin_transaction()
+            cur = self.session.open_cursor(self.uri_prefix + '_0')
```

Comment:
The `cur` is opened without closing the already opened cursor.

Reply:
Nice, fixed it 75c46d1e48ea695bfc07c2eaa341867e7bbc92cd